hybridse/src/udf/udf_library.cc

```diff
@@ -256,29 +256,29 @@ Status UdfLibrary::RemoveDynamicUdf(const std::string& name, const std::vector<n
         lib_name.append(".").append(node::DataTypeName(type));
     }
     if (!HasFunction(canonical_name)) {
-        return Status(kCodegenError, "can not find the function " + canonical_name);
+        return Status(kCodegenError, "can not find the function in udf table: " + canonical_name);
     }
     if (IsUdaf(canonical_name)) {
         std::lock_guard<std::mutex> lock(mu_);
         if (table_.erase(canonical_name) <= 0) {
-            return Status(kCodegenError, "can not find the function " + canonical_name);
+            return Status(kCodegenError, "udaf function not present in udf table: " + canonical_name);
```

Comment:
continue instead of return better ?

Reply:
not hiding but make this function atomic. For udaf there are multiple states to be handled, so either delete nothing, or delete everything.

Reply:
delete may be failed too, there's no everything. You try delele 3 maps, but every map deletetion could fail, I can't see any benifit to change the logic here.